redbot/core/settings_caches.py

```diff
@@ -122,157 +122,145 @@ async def set_ignored_guild(self, guild: discord.Guild, set_to: bool):
+
         else:
             if gid not in self._cached_whitelist:
                 self._cached_whitelist[gid] = await self._config.guild_from_id(gid).whitelist()
             for obj_id in role_or_user:
-                if obj_id in self._cached_whitelist[gid]:
-                    self._cached_whitelist[gid].remove(obj_id)
-                    async with self._config.guild_from_id(gid).whitelist() as curr_list:
-                        curr_list.remove(obj_id)
+                self._cached_whitelist[gid].discard(obj_id)
+            await self._config.guild_from_id(gid).whitelist.set(list(self._cached_whitelist[gid]))
 
-    async def get_blacklist(self, guild: Optional[discord.Guild] = None) -> List[int]:
-        ret: List[int]
+    async def get_blacklist(self, guild: Optional[discord.Guild] = None) -> Set[int]:
+        ret: Set[int]
 
         gid: Optional[int] = guild.id if guild else None
 
         if gid in self._cached_blacklist:
             ret = self._cached_blacklist[gid].copy()
         else:
             if gid is not None:
-                ret = await self._config.guild_from_id(gid).blacklist()
-                if not ret:
-                    ret = []
+                ret = set(await self._config.guild_from_id(gid).blacklist())
             else:
-                ret = await self._config.blacklist()
+                ret = set(await self._config.blacklist())
 
             self._cached_blacklist[gid] = ret.copy()
 
         return ret
 
-    async def add_to_blacklist(self, guild: Optional[discord.Guild], role_or_user: List[int]):
+    async def add_to_blacklist(self, guild: Optional[discord.Guild], role_or_user: Iterator[int]):
         gid: Optional[int] = guild.id if guild else None
         role_or_user = role_or_user or []
-        if not isinstance(role_or_user, list) or not all(
+        if not isinstance(role_or_user, (set, list)) or not all(
             isinstance(r_or_u, int) for r_or_u in role_or_user
         ):
-            raise TypeError("Blacklisted objects must be a list of ints")
+            raise TypeError("Blacklisted objects must be a list or set of ints")
         if gid is None:
             if gid not in self._cached_blacklist:
-                self._cached_blacklist[gid] = await self._config.blacklist()
+                self._cached_blacklist[gid] = set(await self._config.blacklist())
             for obj_id in role_or_user:
-                if obj_id not in self._cached_blacklist[gid]:
-                    self._cached_blacklist[gid].append(obj_id)
-                    async with self._config.blacklist() as curr_list:
-                        curr_list.append(obj_id)
+                self._cached_blacklist[gid].add(obj_id)
+            await self._config.blacklist.set(list(self._cached_blacklist[gid]))
         else:
             if gid not in self._cached_blacklist:
                 self._cached_blacklist[gid] = await self._config.guild_from_id(gid).blacklist()
```

Comment:
You missed a config call here.
```suggestion
                self._cached_blacklist[gid] = set(
                    await self._config.guild_from_id(gid).blacklist()
                )
```